test/integration/expression-tests/array/basic/test.json

```diff
@@ -2,12 +2,13 @@
   "expression": ["array", ["literal", [1, 2, 3]]],
   "inputs": [[{}, {}]],
   "expected": {
-    "outputs": [[1, 2, 3]],
     "compiled": {
       "result": "success",
-      "isZoomConstant": true,
       "isFeatureConstant": true,
+      "isZoomConstant": true,
       "type": "array"
-    }
+    },
+    "outputs": [[1, 2, 3]],
+    "serialized": ["literal", [1, 2, 3]]
```

Comment:
Huh, I'm surprised to see the outer `"array"` assertion missing here -- I don't see any reason over at https://github.com/mapbox/mapbox-gl-native/pull/11156/files#diff-14232703a3132993aadc29177a139a62 why this would happen (and I don't think it would happen on the parsing side either)

Reply:
Oh wait, this is just due to constant folding. 👍